pcdsdevices/ipm.py

```diff
@@ -103,6 +113,13 @@ def removed(self):
         return (self.target.removed and
                 (self.diode.removed or self.diode.inserted))
 
+    def insert(self, moved_cb=None, timeout=None, wait=False):
+        """Move both the target and diode in"""
+        return (self.target.insert(moved_cb=moved_cb, timeout=timeout,
```

Comment:
Can you remind me which target is selected when `insert` is called here? I remember there being four to choose between.

Reply:
Should be the first one.

Reply:
Got it, and I assume it does nothing if we're already in an inserted state?

Reply:
Looks like it doesn't but I feel like it definitely should. This is inherited from InOutPositioner, is there any reason we'd want any other InOutPositioners to not have this behavior?

Reply:
I feel like the desired behavior is:
- always insert the diode
- move the target to T1 if it isn't at a specific target

Reply:
Can you clarify what you mean by a specific target?
> move the target to T1 if it isn't at a specific target

Reply:
`OUT` or `Unknown`

I assume calling `insert` and switching from `T4` to `T1` will cause some confusion

Reply:
Sorry, I'm getting a little confused by what you're saying. Simply, we want the below, right?
```suggestion
        if self.removed:
            return NullStatus()
        return (self.target.insert(moved_cb=moved_cb, timeout=timeout,
```

Reply:
I think the base class is correct to change here, it seems to have always been wrong. This is not the right part change though, we want to hit the `insert` method, not the `inserted` property. Making it look like `remove` is probably correct.